pkg/scheduler/frameworkext/framework_extender.go

```diff
@@ -18,27 +18,48 @@ package frameworkext
 
 import (
 	"context"
+	"fmt"
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/klog/v2"
 	"k8s.io/kubernetes/pkg/scheduler/framework"
 
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
 	koordinatorclientset "github.com/koordinator-sh/koordinator/pkg/client/clientset/versioned"
 	koordinatorinformers "github.com/koordinator-sh/koordinator/pkg/client/informers/externalversions"
+	reservationutil "github.com/koordinator-sh/koordinator/pkg/util/reservation"
 )
 
 var _ FrameworkExtender = &frameworkExtenderImpl{}
 
 type frameworkExtenderImpl struct {
 	framework.Framework
 
+	temporarySnapshot                *TemporarySnapshot
 	sharedListerAdapter              SharedListerAdapter
 	koordinatorClientSet             koordinatorclientset.Interface
 	koordinatorSharedInformerFactory koordinatorinformers.SharedInformerFactory
 
 	preFilterTransformers []PreFilterTransformer
 	filterTransformers    []FilterTransformer
 	scoreTransformers     []ScoreTransformer
+
+	reservationFilterPlugins       []ReservationFilterPlugin
+	reservationScorePlugins        []ReservationScorePlugin
+	reservationRecommender         []ReservationRecommender
```

Comment:
Perhaps `reservationNominator` is better. And we may need to clarify why a ReservationRecommender is needed though we have some ReservationScorePlugins (I think it is like the DefaultBinder for the BindPlugin).

Reply:
Nominator is indeed better. There is a description of why recommender is needed. For details, see the document https://github.com/koordinator-sh/koordinator/pull/1126/files#diff-88ed0b7eb4820fe7359f4af52e61b5fe0e1050c97de1c81790e34e2faf7c7ab3R172

Reply:
OK